components/file_preview/file_preview.tsx

```diff
@@ -1,38 +1,32 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ReactNode} from 'react';
 import {getFileThumbnailUrl, getFileUrl} from 'mattermost-redux/utils/file_utils';
 
 import FilenameOverlay from 'components/file_attachment/filename_overlay.jsx';
 import RemoveIcon from 'components/widgets/icons/fa_remove_icon';
 import Constants, {FileTypes} from 'utils/constants';
 import * as Utils from 'utils/utils.jsx';
 
-import FileProgressPreview from './file_progress_preview.jsx';
+import FileProgressPreview from './file_progress_preview';
 
-export default class FilePreview extends React.PureComponent {
-    static propTypes = {
-        enableSVGs: PropTypes.bool.isRequired,
-        onRemove: PropTypes.func.isRequired,
-        fileInfos: PropTypes.arrayOf(PropTypes.object).isRequired,
-        uploadsInProgress: PropTypes.array,
-        uploadsProgressPercent: PropTypes.object,
-    };
-
-    static defaultProps = {
-        fileInfos: [],
-        uploadsInProgress: [],
-        uploadsProgressPercent: {},
```

Comment:
I think we should keep the `defaultProps`. The parent component is not matched up to respect this component's props yet, so it may not be passing in these props properly.

Reply:
I think these are supposed to be optional anyway

Reply:
restored defaultProps and made properties optional